### PR TITLE
ImagePull failure is now reported on the UI

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
Shows Pod errors clearly on the UI. Also adds a way to add more testcases to demystify pending easily

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
NA

## Tracking Issue
https://github.com/lyft/flyte/issues/244

## Follow-up issue
NA

### CODE REVIEW

File: go/tasks/pluginmachinery/flytek8s/testdata/imagepull-failurepod.json [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+                        "name": "FLYTE_CREDENTIALS_AUTHORIZATION_METADATA_KEY",
+                        "value": "flyte-authorization"
+                    },
+                    {
+                        "name": "FLYTE_CREDENTIALS_SCOPE",
+                        "value": "svc"
+                    },
+                    {
+                        "name": "FLYTE_CREDENTIALS_CLIENT_SECRET",
+                        "value": "E2EeKjJvwKngjrUnmyfvL6tnf8jcPGrVdAxQd8u0"
```
Comment (wild-endeavor) [inline go/tasks/pluginmachinery/flytek8s/testdata/imagepull-failurepod.json:236]: can we clean up some of these things?

Comment (wild-endeavor) [reply go/tasks/pluginmachinery/flytek8s/testdata/imagepull-failurepod.json:236]: i mean this is not really a secret, but still

Comment (kumare3) [reply go/tasks/pluginmachinery/flytek8s/testdata/imagepull-failurepod.json:236]: ohh yes

**Comment by kumare3** [comment]
> and we're not losing anything by no longer showing c.Reason and c.Message right?
Why do you say so. I just suffixed additional information

**Pull request merged**